libsrc/telestrat/cputc.s

```diff
@@ -33,18 +58,37 @@ do_not_change_color_foreground:
 
     stx     OLD_BGCOLOR
 
-    dec     SCRX                 ; Dec SCRX in order to place attribute before the right position
-
     pha
     txa                          ; Swap X to A because, X contains BGCOLOR
-    ORA     #%00010000           ; Add 16 because background color is an attribute between 16 and 23. 17 is red background for example
-    BRK_TELEMON  XFWR            ; Change color on the screen (background)
+    ora     #%00010000           ; Add 16 because background color is an attribute between 16 and 23. 17 is red background for example
+
+    jsr     display_conio
     pla
 
 do_not_change_color:
-    BRK_TELEMON  XFWR            ; Macro send char to screen (channel 0)
+    ; it continues to display_conio
+
+.endproc
+
+.proc display_conio
+    ; This routine is used to displays char on screen
+    ldy     SCRX
+    sta     (ADSCR),y
+    iny
+    cpy     #SCREEN_XSIZE
+    bne     @no_inc
+    ldy     #$00
+    sty     SCRX
+    
+    jmp     update_adscr
+    
+@no_inc:    
+    sty     SCRX
     rts
 .endproc
+
+
+
```

Comment:
In other places you seem to have a horizonal line to separate segments. It would be great if you would keep a consistent style.

Reply:
Fixed